src/Form/WidgetsConfig.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\entity_browser\Form\WidgetsConfig.
+ */
+
+namespace Drupal\entity_browser\Form;
+
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\entity_browser\WidgetManager;
+use Drupal\user\SharedTempStoreFactory;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+class WidgetsConfig extends FormBase {
+
+  /**
+   * Entity browser widget plugin manager.
+   *
+   * @var \Drupal\entity_browser\WidgetManager
+   */
+  protected $widgetManager;
+
+  /**
+   * Tempstore Factory for keeping track of values in each step of the wizard.
+   *
+   * @var \Drupal\user\SharedTempStoreFactory
+   */
+  protected $tempStore;
+
+  /**
+   * WidgetsConfig constructor.
+   * @param \Drupal\entity_browser\WidgetManager $widget_manager
+   * @param \Drupal\user\SharedTempStoreFactory $temp_store
+   * @param null $tempstore_id
+   * @param null $machine_name
+   */
+  function __construct(WidgetManager $widget_manager, SharedTempStoreFactory $temp_store, $tempstore_id = NULL, $machine_name = NULL) {
+    $this->widgetManager = $widget_manager;
+    $this->tempStore = $temp_store;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('plugin.manager.entity_browser.widget'),
+      $container->get('user.shared_tempstore')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'entity_browser_widgets_config_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    /** @var \Drupal\entity_browser\EntityBrowserInterface $entity_browser */
+    $entity_browser = $form_state->getTemporaryValue('wizard')['entity_browser'];
+
+    $widgets = [];
+    foreach ($this->widgetManager->getDefinitions() as $plugin_id => $plugin_definition) {
+      $widgets[$plugin_id] = $plugin_definition['label'];
+    }
+    $default_widgets = [];
+    foreach ($entity_browser->getWidgets() as $widget) {
+      /** @var \Drupal\entity_browser\WidgetInterface $widget */
+      $default_widgets[] = $widget->id();
+    }
+    $form['widget'] = [
+      '#type' => 'select',
+      '#title' => $this->t('Add widget plugin'),
+      '#options' => ['_none_' => '- ' . $this->t('Select a widget to add it') . ' -'] + $widgets,
+      '#ajax' => [
+        'callback' => [get_class($this), 'addWidgetCallback'],
+        'wrapper' => 'widgets',
```

Comment:
It turns out this ID doesn't exist at all. When AJAX framework tries to insert rebuilt table it doesn't know where to put it. 

We either need to make this ID available or change it to something that exists. I propose first option. See further explanation below.